src/main/scala/inox/tip/Parser.scala

```diff
@@ -142,13 +142,22 @@ class Parser(file: File) {
       (Some(tpsLocals.extractTerm(term)), locals)
 
     case DeclareConst(sym, sort) =>
-      (None, locals.withVariable(sym,
-        Variable.fresh(sym.name, locals.extractSort(sort)).setPos(sym.optPos)))
+      val choose = Choose(
+        ValDef(FreshIdentifier(sym.name), locals.extractSort(sort)),
+        BooleanLiteral(true)
+      ).setPos(sym.optPos)
+      (None, locals.withVariable(sym, choose))
```

Comment:
Does this correspond to what you had in mind?

Reply:
I think I would go a `FunDef` instead of variable, to avoid repeating the choose for multiple uses. Something like the `DeclareFun` case.

Reply:
I've updated the code as you suggested but now I've got some other issues. Some test are no longer SAT and some seems to enter an infinite loop or something. Maybe you have some insight on the issue here?

Reply:
Ah, I forgot that free variables in the final expression we want to check are also output as `declare-const` when printing TIP programs... So we should only use the `DeclareFun` trick in the type parametric case and rewrite the benchmarks with `declare-fun` instead of `declare-const` for the match exhaustiveness variables.

However, I don't see any reason for the benchmarks to fail with chooses instead of variables, so it would be useful to debug this anyway!

Reply:
I tried to make the changes but either I did something slightly wrong or there's another bug because it:test gets stuck...